compositor_pipeline/src/queue/audio_queue.rs

```diff
@@ -0,0 +1,101 @@
+use std::{
+    collections::{HashMap, VecDeque},
+    time::{Duration, Instant},
+};
+
+use super::QueueError;
+use compositor_render::{AudioSamplesBatch, AudioSamplesSet, InputId};
+
+#[derive(Debug)]
+pub struct AudioQueue {
+    /// Enqueued sample batches per output
+    input_queues: HashMap<InputId, VecDeque<AudioSamplesBatch>>,
+    /// Stream added timestamp offset in relation to clock start
+    timestamp_offsets: HashMap<InputId, Duration>,
+    /// Stream starting pts offsets (starting pts doesn't have to be 0)
+    start_pts: HashMap<InputId, Duration>,
```

Comment:
you don't need that value

Reply:
I understand those issues, but none of those is an argument to keep those hash maps. All of the problems you listed are easily solvable without hacks involving 2 offsets.